File: android/src/main/java/com/bluetrum/fota/cmd/notification/OtaStateNotificationCallable.java
package com.bluetrum.fota.cmd.notification;

import androidx.annotation.NonNull;

import com.bluetrum.devicemanager.cmd.payloadhandler.PayloadHandler;

public class OtaStateNotificationCallable extends PayloadHandler<Byte> {

    public OtaStateNotificationCallable(@NonNull byte[] payload) {
        super(payload);
    }

    @Override
    public Byte call() throws Exception {
        final byte[] payload = getPayload();
        if (payload.length == 1) {
            return payload[0];
        }
        return null;
    }

}

File: ios/Classes/FOTA/Command/Notification/OtaStateNotificationCallable.swift
//
//  OtaStateNotificationCallable.swift
//  FOTA
//
//  Created by Bluetrum.
//  

import Foundation
import DeviceManager

public final class OtaStateNotificationCallable: AbstractPayloadHandler {
    
    public override func callAsFunction() -> AnyObject? {
        if payload.count == 1 {
            return payload[0] as AnyObject
        }
        return nil
    }
    
}
